src/Widgets/SearchItem.vala

```diff
@@ -69,12 +69,19 @@ namespace Slingshot.Widgets {
             var uri_match = app.match as Synapse.UriMatch;
             if (uri_match != null && uri_match.uri.has_prefix ("http")) {
                 cancellable = new Cancellable ();
-                Backend.SynapseSearch.get_favicon_for_match.begin (uri_match,
-                    ICON_SIZE, cancellable, (obj, res) => {
-
+                Backend.SynapseSearch.get_favicon_for_match.begin (uri_match, ICON_SIZE, cancellable, (obj, res) => {
                     var pixbuf = Backend.SynapseSearch.get_favicon_for_match.end (res);
-                    if (pixbuf != null)
+                    if (pixbuf != null) {
                         icon.set_from_pixbuf (pixbuf);
+                    }
+                });
+            } else if (app.match != null && app.match.icon_name.has_prefix ("/")) {
```

Comment:
Use `...has_prefix (Path.DIR_SEPARATOR_S);`
Use `uri_match` here since the `app.match` was already assigned to it.